Backend/Dockerfile

```diff
@@ -1,36 +1,48 @@
-# Rust version 
-FROM rust:1.69.0 as build
-
-# create a new empty shell project
-# RUN USER=root cargo new --bin password_manager
-WORKDIR /password_manager_webapp
-
-# Install the required system dependencies for our linking configuration
+# prepare chef reciepe
+FROM lukemathwalker/cargo-chef:latest-rust-1.69.0 as chef
+WORKDIR /password_manager
 RUN apt update && apt install lld clang -y
-
-# copy the files in local machine from the docker image
+FROM chef as planner
 COPY . .
-
-# set SQLX_OFFLINE var to true
+RUN cargo chef prepare --recipe-path recipe.json
+
+# Builder stage
+FROM chef as builder
+COPY --from=planner /password_manager/recipe.json recipe.json
+# Build out project dependecies, not our application!
+RUN cargo chef cook --release --recipe-path recipe.json
+# upto this point, if our dependecy tree stays the same
+# all layers should be cached.
+COPY  . .
+# set env var sql_offline to TRUE
 ENV SQLX_OFFLINE true
-
-# Build the binary in --release mode
-RUN cargo build --release
+# build in relase mode
+RUN cargo build --release --bin password_manager_webapp
+# clean build environment
 RUN rm src/*.rs
 
 # Now that the dependency is built, copy your source code
 COPY ./src ./src
-
 #  build for release
 RUN rm ./target/release/deps/password_manager_webapp*
 RUN cargo build --release
 
-# final base
```

Comment:
use 'FROM rust:1.59.0-slim AS runtime' to make the image even smaller than 100mb